operator/cmd/mesh/install.go

```diff
@@ -348,3 +366,75 @@ func getProfileNSAndEnabledComponents(iop *v1alpha12.IstioOperator) (string, str
 	}
 	return iop.Spec.Profile, name.IstioDefaultNamespace, enabledComponents, nil
 }
+
+// requiresIstiodServiceCreation determines if the CP requires an istiod service to function
+func requiresIstiodServiceCreation(cs kubernetes.Interface, istioNs string) bool {
+	// First: does the istiod service exist?
+	_, err := cs.CoreV1().Services(istioNs).Get(
+		context.TODO(), "istiod", metav1.GetOptions{})
+	if !kerrors.IsNotFound(err) {
+		vwh, err := cs.AdmissionregistrationV1().ValidatingWebhookConfigurations().Get(
+			context.TODO(), fmt.Sprintf("istiod-%s", name.IstioDefaultNamespace), metav1.GetOptions{})
+		if err != nil {
+			fmt.Printf("oops: %v", err)
+			return false
+		}
+		// Second: if it doesn't, are there webhooks in the VWHC referencing it? If so,
+		// we should create to avoid broken validation
+		for _, wh := range vwh.Webhooks {
+			if wh.ClientConfig.Service != nil {
+				if wh.ClientConfig.Service.Name == "istiod" {
+					return true
+				}
+			}
+		}
+	}
+
+	return false
+}
+
+// createIstiodService creates an `istiod` service pointed at the given revision
+func createIstiodService(cs kubernetes.Interface, istioNs, revision string) error {
+	const (
+		pilotDiscoveryChart = "istio-control/istio-discovery"
+		serviceTemplateName = "service.yaml"
+	)
+	r := helm.NewHelmRenderer("", pilotDiscoveryChart, "Pilot", istioNs)
+
+	if err := r.Run(); err != nil {
+		return fmt.Errorf("failed running Helm renderer: %v", err)
+	}
+
+	values := fmt.Sprintf(`
+revision: "%s"
+`, revision)
```

Comment:
if you don't put revision here, perhaps you don't need to do the rename around line 433?

Reply:
Problem there is that we'd to change the service selectors to point to the revision (so we have to change something either way)

Reply:
I see, thanks for the clarification